broker/lib/fabrik/DockerInstance.js

```diff
@@ -245,6 +245,7 @@ class DockerInstance extends BaseInstance {
           .startContainer()
           .catch(DockerError.ServerError, err => self
             .removeContainer()
+            .then(() => docker.updatePortRegistry())
```

Comment:
Dont you get a specific error in case of port already in use? Ideally we want to updatePortRegistry only on that scenario and not all error scenarios right?

Reply:
As Jagadish said, if possible, we can update cache on specific issue.
Also, this means, the create service instance still fails first time and next time onwards, it will not happen. Not sure if this is acceptable.

Reply:
I tested and the port already in use issue comes as 500 and start api of docker throws following codes
    statusCodes: {
      200: true, // unofficial, but proxies may return it
      204: true,
      304: 'container already started',
      404: 'no such container',
      500: 'server error'
    },

Reply:
I am ok, with the above in case we are not getting specific error codes. Approving this PR.

Reply:
Also, as we retry 3 times to create container, it will not be create instance failure from user point of view. Hence, sounds good. Approving PR.